rpc/src/services/stream_services.rs

```diff
@@ -0,0 +1,370 @@
+                        let head_string_result = self.yield_head(Some(current_head));
+                        Poll::Ready(head_string_result.transpose())
+                    }
+                } else {
+                    // No current head found, storage not ready yet
+                    Poll::Ready(None)
+                }
+            }
+        }
+    }
+}
+
+impl Stream for OperationMonitorStream {
+    type Item = Result<String, serde_json::Error>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Result<String, serde_json::Error>>> {
+        // create or get a delay future, that blocks for MONITOR_TIMER_MILIS
+        let delay = self.delay.get_or_insert_with(|| {
+            let when = Instant::now() + Duration::from_millis(MONITOR_TIMER_MILIS);
+            delay_until(when)
+        });
+
+        // pin the future pointer
+        let mut pinned = std::boxed::Box::pin(delay);
+        let pinned_mut = pinned.as_mut();
+
+        // poll the delay future
+        match pinned_mut.poll(cx) {
+            Poll::Pending => {
+                Poll::Pending
+            },
+            _ => {
+                // get rid of the used delay
+                self.delay = None;
+
+                let state = self.state.read().unwrap();
+                let current_head = state.current_head().clone();
+
+                // drop the immutable borrow so we can borrow self again as mutable
+                // TODO: refactor this drop (remove if possible)
+                drop(state);
+
+                let last_checked_head = if let Some(head_hash) = &self.last_checked_head {
+                    head_hash
+                } else {
+                    return Poll::Ready(None)
+                };
+
+                if let Some(current_head) = current_head {
+                    if last_checked_head == &current_head.header().hash {
+                        // current head not changed, check for new operations
+                        let yielded = self.yield_operations();
+                        match yielded {
+                            Poll::Pending => {
+                                cx.waker().wake_by_ref();
+                                Poll::Pending
+                            },
+                            _ => {
+                                yielded
+                            },
+                        }
+                    } else {
+                        // Head change, end stream
+                        Poll::Ready(None)
+                    }
+                } else {
+                    // No current head found, storage not ready yet 
+                    Poll::Ready(None)
+                }
+            }
+        }
+    }
+}
+
+/// Get information about current head monitor header as a stream of Json strings
+pub(crate) fn get_current_head_monitor_header(chain_id: &ChainId, env: &RpcServiceEnvironment, protocol: Option<String>) -> Option<HeadMonitorStream> {
```

Comment:
@adonagy 
one more thing, change pls. this:
`protocol: Option<String>` -> `protocol: Option<ProtocolHash>`